src/main/java/net/neoforged/neoforge/common/NeoForgeEventHandler.java

```diff
@@ -133,7 +133,8 @@ private <T> void handleSync(ServerPlayer player, Registry<T> registry, Collectio
             if (attach == null || attach.networkCodec() == null) return;
             att.put(key, registry.getDataMap(attach));
         });
-        if (!att.isEmpty()) {
+        // Note: don't send data maps over in-memory connections, else the client-side handling will wipe non-synced data maps.
+        if (!att.isEmpty() && !player.connection.getConnection().isMemoryConnection()) {
```

Comment:
Do we even need to bother populating the `att` map for a memory connection? It seems to just get thrown away immediately.

Reply:
Moved the check earlier.